contracts/actions/yearn/YearnSupply.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "../ActionBase.sol";
+import "../../utils/TokenUtils.sol";
+import "../../interfaces/yearn/yVault.sol";
+import "../../interfaces/yearn/YearnRegistry.sol";
+import "../../DS/DSMath.sol";
+
+/// @title Supplies tokens to Yearn vault
+contract YearnSupply is ActionBase, DSMath {
+    using TokenUtils for address;
+
+    YearnRegistry public constant yearnRegistry = 
+        YearnRegistry(0x50c1a2eA0a861A967D9d0FFE2AE4012c2E053804);
+
+    /// @param token - address of token to supply
+    /// @param amount - amount of token to supply
+    /// @param from - address from which to pull tokens from
+    /// @param to - address where received yTokens will be sent to
+    struct Params {
+        address token;
+        uint256 amount;
+        address from;
+        address to;
+    }
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory inputData = parseInputs(_callData);
+
+        inputData.amount = _parseParamUint(inputData.amount, _paramMapping[0], _subData, _returnValues);
+        inputData.from = _parseParamAddr(inputData.from, _paramMapping[1], _subData, _returnValues);
+        inputData.to = _parseParamAddr(inputData.to, _paramMapping[2], _subData, _returnValues);
+        
+        uint256 yAmountReceived = _yearnSupply(inputData);
+        return bytes32(yAmountReceived);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public payable override {
+        Params memory inputData = parseInputs(_callData);
+
+        _yearnDirectSupply(inputData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @dev this function sends only the amount of yTokens received from supplying to an account
+    /// @return yTokenAmount amount of yTokens we received for supplying
+    function _yearnSupply(Params memory _inputData) internal returns (uint256 yTokenAmount) {
+        YVault vault = YVault(yearnRegistry.latestVault(_inputData.token));
+
+        uint amountPulled = _inputData.token.pullTokensIfNeeded(_inputData.from, _inputData.amount);
```

Comment:
If amount uint.max pull whole balance _from?

Reply:
yes, tokenUtils handles uint_max and puts whole balance as return value